consensus/src/quorum_store/quorum_store_coordinator.rs

```diff
@@ -54,17 +54,25 @@ impl QuorumStoreCoordinator {
             monitor!("quorum_store_coordinator_loop", {
                 match cmd {
                     CoordinatorCommand::CommitNotification(block_timestamp, batches) => {
+                        // TODO: need a callback or not?
```

Comment:
callback for what?

Reply:
There's a race here. If batch generator's commit update takes place before proof coordinator, we could recreate a new batch with the same batch digest in batch generator and then votes conflict with the known batch in digest_to_proof.

In practice, creating the same batch should be rare, and doing so while proof coordinator's commit is delayed (vs votes) should be even more rare. In forge I didn't see this race ever kicking in.

But if we were to want to guarantee order here, we'd have to wait for a callback from each component.

Reply:
what happens if the race does occur? does a node permanently get stuck with being able to create batches, or?

Reply:
Just the batch with the retried txns has a chance it won't reach quorum. It will only affect that batch until it times out

Reply:
sounds good, seems insignificant